src/common-types.d.ts

```diff
@@ -125,6 +125,221 @@ interface Coloring {
 	info?: boolean;
 }
 
+interface ThemeColorSet {
+	main: string;
+	light: string;
+	dark: string;
+}
+
+interface WithSemilight {
+	semilight: string;
+}
+
+type ColorShade = keyof ThemeColorSet;
+
+type Shading = { shade?: ColorShade };
+
 interface Sizing {
 	emphasized?: boolean;
 }
+
+type TooltipPlacement = 'top' | 'right' | 'bottom' | 'left';
+
+interface TooltipProps {
+	text: string;
+	trigger?: 'click' | 'hover';
+	placement?: TooltipPlacement;
+	containerStyle?: React.CSSProperties;
+	innerStyle?: React.CSSProperties;
+	arrowStyle?: React.CSSProperties;
+}
+
+/* Used by:
+	1. CodeWithCopy.tsx
+	2. DropdownButton.tsx
+	3. Link.tsx
+	4. Txt.tsx
+*/
+interface Tooltip {
+	tooltip?: string | TooltipProps;
+}
+
+/* Used by:
+	1. ArcSlider.tsx
+	2. Badge.tsx
+	3. Grid.tsx
+	4. 'extra' & 'unstable' components
+*/
+interface BoxProps extends DefaultProps, Tooltip {
+	flex?: string | string[];
+	order?: number | string | Array<number | string>;
+}
+
+/* Used by:
+	1. Txt.tsx
+	2. Pill.tsx
+*/
+interface TxtProps extends DefaultProps, Tooltip {
```

Comment:
I am commenting on the `TxtProps`, but most of these "common types" are not actually necessary. What you need to do is export the interface from the component (for example, `export interface TxtProps` in `TxtProps`, and then import it from there in the `Pill` component in this case.

Reply:
makes sense @sradevski Thanks for the pointer

Reply:
I tried it and it looks good, but my problem lies with the 'extra.d.ts' and 'unstable.d.ts' files, where I can't reference these typings relatevely.

I might have to break those apart.